### DISCUSSION TITLE
.q-tooltip--style selector doesn't work. WHY?

### DISCUSSION THREAD

**Discussion by quroom** [quasarframework/quasar#16350] [General - Components / Directives / etc] [answered]
I add selector in style scope.

.q-tooltip--style {
  background: red !important;
}

It doesn't work.
I don't have any idea.

**Comment by quroom** [comment, marked as answer]
https://stackoverflow.com/questions/48032006/how-do-i-use-deep-or-or-v-deep-in-vue-js

:global(.q-tooltip--style) {
   background: red !important;
}
This selector works.
But use it carefully. Because it can make side effects to another component.

**Reply by yusufkandemir** [reply]
Use `:deep` instead of `:global` to avoid styles from leaking to other components.
https://vuejs.org/api/sfc-css-features.html#deep-selectors

**Reply by quroom** [reply]
I already tried it but it doesn't work.
Maybe tooltip code isn't injected in page or component code.
It's injected in body section like dialog..

**Reply by yusufkandemir** [reply]
Yes, you are correct. You can give q-tooltip a class of your choice, and use it in the selector to avoid affecting other components.
```html
<q-tooltip class="danger-tooltip" />

<style lang="scss" scoped>
:global(.danger-tooltip .q-tooltip--style) {
  background: red !important;
}
</style>
```

By the way, if it's just for making the background red, you can simply use `class="bg-red"`.

**Reply by quroom** [reply]
I wanted font-size unset.
Butcause of font-size is little small in desktop.
So I would like to make it bigger by user input.

I am sure adding class is better solution like you mentioned.
But I want to make more clear.

There is no spaces between class selectors.

```
<q-tooltip class="unset-font"/>
```

```
<style scoped>
:global(.unset-font.q-tooltip--style) {
  font-size: unset;
}
</style>
```

**Reply by yusufkandemir** [reply]
Yeah, you are right again 👍 _I need to clear my mind._

**Reply by quroom** [reply]
Thanks for kind answer :)
Happy coding 👍
